### DISCUSSION TITLE
Bluetooth on Rpi 4 model B not working

### DISCUSSION THREAD

**Discussion by masterx1981** [fl4p/batmon-ha#161] [General]
Hi! Sorry, but i'm fairly new on home assistat/rpi hw... I'm trying to get this integration work with 2 ant bms, 32s in series (already changhed mac address on them). The ant bms works (almost) as it should.
I've haasio installed and updated. I can't get the pi to see the 2 bt devices. I see in the devices the mac address of the antenna, but running bluetoothctl devices or scan on it not find anything.
This is the cfg:
- address: 32:E3:2E:01:00:01
  type: ant
  alias: Pacco 1
  adapter: hci0
  debug: false
  current_calibration: 1
- address: 32:E3:2E:01:00:02
  type: ant
  alias: Pacco 2
  adapter: hci0
  debug: false
  current_calibration: 1

The 2 mac are correct. But in the log i get 
02:30:56 ERROR [sampling] Pacco 2 error: Device 32:E3:2E:01:00:02 not discovered. Make sure it in range and is not being controled by another application. (set())
And
02:30:35 ERROR [main] Error (num 3, max 200) reading BMS: Device 32:E3:2E:01:00:01 not discovered. Make sure it in range and is not being controled by another application. (set())

Any suggestion?
Thanks!

**Comment by fl4p** [comment]
Hi,
try 
```
bluetoothctl power off
bluetoothctl power on
```
and disable bluetooth on all nearby phones, computers, tablets.

**Reply by masterx1981** [reply]
Nothing seem work, the bt never find anything. Also rebooted the pi hw. Under the "settings" "Devices and services" "bluetooth" i find one integration voice, with a mac address, if i click on it i get hci0(XX:XX:XX:XX:XX:XX), connected. 
But if i go under "settings" ""hardware" "all the hardware" i not see the hci0 device.
Thanks for the help!

**Comment by fl4p** [comment]
can you run it outside the docker container directly from the shell?

**Reply by masterx1981** [reply]
Uhm, sorry but i'm not sure how i can do it. I've installed the advanced ssh and web terminal addon and i launch the commands both from the ssh and from the web terminal.

**Reply by fl4p** [reply]
what do you see under http://homeassistant.local:8123/config/integrations/integration/bluetooth ?

**Comment by masterx1981** [comment]
![Screenshot_20231122_202831_Chrome](https://github.com/fl4p/batmon-ha/assets/76602320/a49a4a0b-84f1-46b2-8efc-97f586ec03f3)
![Screenshot_20231122_202756_Chrome](https://github.com/fl4p/batmon-ha/assets/76602320/7d6cca82-2c07-4d80-a156-c069b610284e)

**Comment by fl4p** [comment]
disable the bluetooth integration and restart batmon (it doesnt need the integration as it accesses the bt adapter)

**Comment by masterx1981** [comment]
Nothing, same as before...
![Screenshot_20231122_231710_Home Assistant](https://github.com/fl4p/batmon-ha/assets/76602320/74215b89-83ae-4023-9a5f-f14cb652f4a9)
![Screenshot_20231122_232017_Home Assistant](https://github.com/fl4p/batmon-ha/assets/76602320/7eb2d235-fc4d-4ef3-8ea4-7e296995fae6)

**Comment by masterx1981** [comment]
The 2 bms are at 50cm from the raspberry pi, and the phone see them quite well
![Screenshot_20231122_232143_BLE Scanner](https://github.com/fl4p/batmon-ha/assets/76602320/6be1cdc6-c104-4922-a6bf-21e54a021ba2)

**Comment by masterx1981** [comment]
Little addendum. In the supervisor logs i get a lot of:
`23-11-23 01:34:40 WARNING (MainThread) [supervisor.addons.options] Unknown option 'adapter' for Batmon (2af0a32d_batmon)`
But i not think that is a batmon proble, as also enabling the bt le tracker, it seem to not find any device.

**Comment by 7wells** [comment]
@masterx1981 
Usually, BT works out of the box on Home Assistant (HA). Does your RPi 4 show any other BT devices? How did you install HA, i.e. which guide did you follow for its installation? Please show the result of the command `bluetoothctl`. You can enter it from an SSH session or directly from within HA - official "Terminal & SSH" add-on or "Advanced SSH & Web Terminal" community add-on. You might want to consider redacting any MAC addresses that should pop up after issuing this command to keep your privacy.

PS: If I remember correctly, poor BT connectivity from an RPi 4 can be overcome by switching its wifi off. Then, of course, you would have to connect your RPi 4 via ethernet cable. HTH!

**Reply by masterx1981** [reply]
Hi!
At the end i gave up on bluetooth, and installed an esp01 on the bms, and seem to work fine, with the benefit of the longer range.
The problem was that, also with the discovery enabled, i've never seen any bt device. Maybe my board have a fault on the bt module.

**Comment by 6bishnoi** [comment]
same problem for me, try to disable the rpi4 wifi
how do you installed esp01 ?

**Reply by masterx1981** [reply]
I've used this :
https://github.com/syssi/esphome-ant-bms
via the home assistant esphome integration, now it's connected via wifi, and work very well.

**Comment by 6bishnoi** [comment]
oh, I have jkbms !
